lib/db/queries.ts

```diff
@@ -37,3 +37,13 @@ export const deleteScenario = async (id: number) => {
     const res = await db.delete(ScenariosTable).where(eq(ScenariosTable.id, id)).returning();
     return res.map((row) => ({ ...row, data: scenarioSchema.parse(JSON.parse(row.data)) }));
 };
+
+export const getOrInsertUser = async (id: string) => {
+    const res = await db.query.UsersTable.findFirst({ where: (user, { eq }) => eq(user.id, id) });
+    return res || (await db.insert(UsersTable).values({ id }).returning()).find(() => true);
```

Comment:
maybe 

```suggestion
    return res || (await db.insert(UsersTable).values({ id }).returning()).at(0);
```

instead?

Reply:
I LIKE! Thanks